### PR TITLE
(BOLT-794) Adjustments to sensitive handling

### PR DESCRIPTION
**Pull request by nmaludy**
### Changes

* Plans can no longer pass in `Sensitive` typed arguments
* `run_task()` validates input parameters against the `Data` type once again
* The `Boltlib::ArgsSpec` type has been removed, since `Sensitive` is no longer accepted as an argument type.
* The `Sensitive` type is no longer listed in the `writing_tasks` documentation for acceptable parameter data types

### Remains the same

* Task authors can still specify `"sensitive": true` on parameters and they will be wrapped with `Sensitive` under the hood and not logged (except in `--debug` mode).

### CODE REVIEW

File: pre-docs/writing_tasks.md [hunk 1]
```diff
@@ -628,25 +624,10 @@ There are two options for defining a task parameter as `Sensitive`:
...
-```
-
-When a parameter is `Sensitive` the only expectation, currently, is that the parameter
-will not be logged. No other precautions are taken.
+When a parameter is `sensitive` the only expectation, currently, is that the parameter
+will not be logged (except when passing `--debug`). No other precautions are taken.
 
-It is up to the task author to implement their tasks in such a way that `Sensitive`
+It is up to the task author to implement their tasks in such a way that `sensitive`
 parameters are not exposed or logged. Please be mindful when coding your tasks!
```
Comment (MikaelSmith) [inline pre-docs/writing_tasks.md]: Can you also remove line 706/687, the entry about `Sensitive`? I think that was added accidentally, and doesn't work.

File: pre-docs/writing_plans.md [hunk 2]
```diff
@@ -292,7 +292,7 @@ $r.each |$result| {
 
 ## Passing sensitive data to tasks
 
-Tasks can define parameters to be `Sensitive`, meaning the parameter values will not be logged by Bolt.
+Tasks can define parameters to be `sensitive`, meaning the parameter values will not be logged by Bolt.
```
Comment (MikaelSmith) [inline pre-docs/writing_plans.md]: Should we note somewhere that you need to explicitly unwrap `Sensitive` types before passing to `run_task`?

**Comment by nicklewis** [comment]
This works exactly how I was thinking of. :+1:

File: pre-docs/writing_plans.md (hunk 2 above)
Comment (nmaludy) [reply pre-docs/writing_plans.md]: Added a sub-section below this.

**Comment by adreyer** [comment]
Can you squash this into a single commit

**Pull request merged**